mobile/library/common/jni/types/exception.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <string>
+
+#include "library/common/jni/import/jni_import.h"
+
+namespace Envoy {
+namespace JNI {
+
+/**
+ * @brief A convenience wrapper for checking for and reporting JNI exceptions.
+ */
+class Exception {
+public:
+  /**
+   * @brief Checks and clears any pending exceptions. Reports pending exceptions to a platform
+   * layer.
```

Comment:
worth documenting how it's reported?  AFIK really it just logs a message which may or may not be intercepted at the platform layer

Reply:
added more details